### DISCUSSION TITLE
RSSI value from the clients

### DISCUSSION THREAD

**Discussion by aaflatooni** [RaspAP/raspap-webgui#1079] [General]
I am looking to monitor the signal strength for the clients connected to my RaspAP.

Is there a way to get the RSSI (signal strength) from the connected clients?
Thanks

**Comment by billz** [comment]
Good question. `iw` is used to get info about clients (stations) connected to the AP:

```
 $ iw dev wlan0 station dump
Station f4:00:00:00:00:00 (on wlan0)
	inactive time:	19000 ms
	rx bytes:	353772
	rx packets:	1908
	tx bytes:	8213711
	tx packets:	39672
	tx failed:	0
	tx bitrate:	65.0 MBit/s
	rx bitrate:	1.0 MBit/s
	authorized:	yes
	authenticated:	yes
	associated:	yes
	WMM/WME:	no
	TDLS peer:	yes
	DTIM period:	2
	beacon interval:100
	short slot time:yes
	connected time:	772 seconds
	current time:	18446744072154520352 ms
```

RSSI is not returned here, nor when getting a specific station, eg. `iw dev wlan0 station get [mac]`.

However `sudo iw dev wlan0 scan` does return signal values for STAs. This could be used to add RSSI to a client list. The only downside is the scan takes quite awhile to execute; in this case ~4000ms.

**Reply by aaflatooni** [reply]
Thanks,
Is there a way to speed up the scan or that is the timeout interval for all the clients to respond back?

**Reply by billz** [reply]
iw is well documented, check `iw help`. You could also try `iwlist [interface] scanning [essid NNN]`
